src/qml/controls/BlockClock.qml

```diff
@@ -0,0 +1,160 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+Item {
```

Comment:
This is the implementation of the block clock, and the file residing at `src/qml/components/BlockClockComponent.qml` is just a controller for this. A better architecture is to have a controller that loads the different states the block clock can be in.

You'd be able to split up this file, as it is very long, between a simple circular progress bar that shows the verification progress while in IBD, then another dial like progress bar for the blocks. Then load these into the controller.

Reply:
Let me try rephrasing your suggestion to ensure I understood it correctly.
You suggest I create a separate `component` file for each block clock state, create a controller file in `qml/control`, and switch to loading different block clock component files based on the requirement.

Have I understood it correctly?